autoscheduler/frontend/src/components/SchedulingPage/Schedule/Schedule.tsx

```diff
@@ -465,17 +504,24 @@ const Schedule: React.FC = () => {
     };
 
     throttle(`${term}`, saveAvailabilities, 3000, true);
-  }, [availabilityList, term, isMouseDown, isLoadingAvailabilities]);
+  }, [availabilityList, term, isMouseDown, isLoadingAvailabilities, screenshot]);
 
   // On unmount, force-call the previously called throttle functions
   // This way when we navigate back to the homepage we can guarantee saveAvailabilities
   // will have been called
   React.useEffect(() => (): void => {
+    if (screenshot) return;
+
     throttle('', () => {}, 2 ** 31 - 1, true);
-  }, []);
+  }, [screenshot]);
+
+  // The screenshottable-schedule does not render corrctly if isLoadingAvailabilities == true
+  React.useEffect(() => {
+    if (screenshot) setIsLoadingAvailabilities(false);
+  }, [screenshot]);
```

Comment:
```ts
const [isLoadingAvailabilities, setIsLoadingAvailabilities] = React.useState(!screenshot);
```
Isn't it a better idea to set the initial state correctly like above rather than set it in a hook?

Reply:
Yeah brainfart lol